.github/workflows/ci.yml

```diff
@@ -273,57 +269,35 @@ jobs:
             gcr.io/iree-oss/base@sha256:5d43683c6b50aebe1fca6c85f2012f3b0fa153bf4dd268e8767b619b1891423a \
             ./build_tools/cmake/build_runtime.sh \
             "${BUILD_DIR}"
-      # Using a tar archive is necessary to preserve file permissions. See
-      # https://github.com/actions/upload-artifact#maintaining-file-permissions-and-case-sensitive-files
-      # The upload action already does its own gzip compression, so no need to
-      # do our own.
-      - name: "Create build dir archive"
-        run: tar -cf ${BUILD_DIR}.tar ${BUILD_DIR}
-      - uses: actions/upload-artifact@3cea5372237819ed00197afe530f5a7ea3e805c8 # v3.1.0
-        with:
-          name: "${{ env.BUILD_DIR }}.tar"
-          path: "${{ env.BUILD_DIR }}.tar"
+      - name: "Testing runtime"
+        run: |
+          ./build_tools/github_actions/docker_run.sh \
+            --env IREE_VULKAN_DISABLE=1 \
+            gcr.io/iree-oss/base@sha256:5d43683c6b50aebe1fca6c85f2012f3b0fa153bf4dd268e8767b619b1891423a \
+            ./build_tools/cmake/ctest_all.sh \
+            "${BUILD_DIR}"
```

Comment:
Maybe make this a single `docker run` call as in our other build&test workflows

Reply:
It's not just characters repeating: it's also tearing down and relaunching the container (although I'm not sure how long that takes, in practice. Apparently less than 3s). I've steered away from GitHub actions specific ways of using Docker containers because they're more tied to the CI and therefore harder to reproduce locally or migrate to other CI systems. That includes translating all the `docker run` options into the GitHub Actions syntax. The script for doing this does additional setup (like creating directories to mount), that would have to be sharded out elsewhere. Overall, it just seems like a false simplification. One option we have would be to start the container running in one step and then use `docker exec` to run various commands in it.

Regardless, what you've got here is fine for now, though I do still have a slight preference for doing it in a single step to match how other jobs work and avoid launching a container multiple times (although that doesn't seem to have meaningful latency overhead). If you wanted to follow-up by splitting all of them, then that would be fine too :-P